http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/HttpOrHttp2ChannelPool.java

```diff
@@ -171,7 +171,22 @@ private void release0(Channel channel, Promise<Void> promise) {
 
     @Override
     public void close() {
-        doInEventLoop(eventLoop, protocolImpl::close);
+        doInEventLoop(eventLoop, this::close0);
     }
 
+    private void close0() {
+        if (protocolImplPromise == null) {
```

Comment:
Seems like `failProtocolImplPromise` would null `protocolImplPromise` out, do we need to close the channel pool there?

Reply:
Hmm I don't think so; if `protocolImplPromise` is null then we didn't create `protocolImpl` so closing `delegatePool` directly would be the right thing to do I think. Also the pool is still technically open at that point; the caller could call `acquire` again, and it could succeed this time.